### DISCUSSION TITLE
Accessing IDs and Logs via API in Prefect

### DISCUSSION THREAD

**Discussion by jayashriv710** [PrefectHQ/prefect#15179] [Q&A]
Hi Prefect Team,

I am currently working on integrating Prefect into our workflow and would like to inquire about accessing specific data programmatically. Specifically, I am interested in retrieving the following IDs:

- Flow IDs
- Flow Run IDs
- Task Run IDs

Additionally, I would like to know if it is possible to access logs through the API.

Could you please provide guidance on how to achieve this using the Prefect API? Any relevant documentation or examples would be greatly appreciated.

Thank you in advance :)

**Comment by cicdw** [comment]
Hi @jayashriv710 - yes all of this is possible; Prefect exposes a REST API that is [documented here](https://docs.prefect.io/3.0/api-ref/rest-api/index).  Is there something missing from that documentation that would be helpful to your use case, or does that cover your needs?
